packages/flame/test/gestures/detectors_test.dart

```diff
@@ -30,6 +32,1022 @@ void main() {
+          ),
+        );
+
+        expect(game.forcePressStart, equals(1));
+        expect(game.forcePressPeaked, equals(1));
+        expect(game.forcePressUpdate, equals(2));
+        expect(game.forcePressEnded, equals(0));
+
+        await gesture.up();
+
+        expect(game.forcePressStart, equals(1));
+        expect(game.forcePressPeaked, equals(1));
+        expect(game.forcePressUpdate, equals(2));
+        expect(game.forcePressEnded, equals(1));
+      },
+    );
+
+    testWithGame<_ForcePressDetectorGame>(
+      'can be Force Press started',
+      _ForcePressDetectorGame.new,
+      (game) async {
+        await game.ready();
+
+        game.handleForcePressStart(
+          ForcePressDetails(
+            globalPosition: const Offset(10, 10),
+            pressure: .4,
+          ),
+        );
+
+        expect(game.forcePressStart, equals(1));
+      },
+    );
+    testWithGame<_ForcePressDetectorGame>(
+      'can be Force Press Updated',
+      _ForcePressDetectorGame.new,
+      (game) async {
+        await game.ready();
+
+        game.handleForcePressUpdate(
+          ForcePressDetails(
+            globalPosition: const Offset(10, 10),
+            pressure: .7,
+          ),
+        );
+
+        expect(game.forcePressUpdate, equals(1));
+      },
+    );
+    testWithGame<_ForcePressDetectorGame>(
+      'can be Force Press peaked',
+      _ForcePressDetectorGame.new,
+      (game) async {
+        await game.ready();
+
+        game.handleForcePressPeak(
+          ForcePressDetails(
+            globalPosition: const Offset(10, 10),
+            pressure: .9,
+          ),
+        );
+
+        expect(game.forcePressPeaked, equals(1));
+      },
+    );
+    testWithGame<_ForcePressDetectorGame>(
+      'can be Force Press Ended',
+      _ForcePressDetectorGame.new,
+      (game) async {
+        await game.ready();
+
+        game.handleForcePressEnd(
+          ForcePressDetails(
+            globalPosition: const Offset(10, 10),
+            pressure: .2,
+          ),
+        );
+        expect(game.forcePressEnded, equals(1));
+      },
+    );
+  });
+
+  group('PanDetector', () {
+    final panGame = FlameTester(_PanDetectorGame.new);
+
+    panGame.testGameWidget(
+      'can Register pan',
+      verify: (game, tester) async {
+        await tester.dragFrom(const Offset(10, 10), const Offset(20, 20));
+
+        expect(game.hasPanStart, isTrue);
+        expect(game.hasPanDown, isTrue);
+        expect(game.hasPanUpdate, isTrue);
+        expect(game.hasPanEnd, isTrue);
+      },
+    );
+    // Todo(dipak) : should register pan cancel.
+    panGame.testGameWidget(
+      'can Register pan cancel',
+      verify: (game, tester) async {
+        final gesture = await tester.startGesture(const Offset(10, 10));
+
+        await gesture.up();
+
+        // expect(game.hasPanStart, isTrue);
+        // expect(game.hasPanDown, isTrue);
+        // expect(game.hasPanCancel, isTrue);
```

Comment:
For a pan to cancel another gesture must win the arena after the pan has started, I'm not sure how to trigger that.